### PR TITLE
Initial build of components and setup

### PR DESCRIPTION
**Pull request by kavanparam**
Still work to be done including:
- Dice component — add random numbers appear in dice component & make roll dice button functional
- Grid for gameboard

image

### CODE REVIEW

File: components/math/multiplicationConnect/DiceSection.tsx
Comment (vithushan19) [inline components/math/multiplicationConnect/DiceSection.tsx:22]: I know this is a work in progress, but you shouldn't include commented code in your PRs.

Can we remove this?

File: components/math/multiplicationConnect/PlayerSection.tsx
Comment (vithushan19) [inline components/math/multiplicationConnect/PlayerSection.tsx:4]: Nice!

File: pages/studentPortal/labs/[courseId]/index.tsx
Comment (vithushan19) [inline pages/studentPortal/labs/[courseId]/index.tsx:36]: Awesome!!

File: pages/studentPortal/labs/multiplication-connect/Index.tsx
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication-connect/Index.tsx:1]: This doesn't belong here

File: pages/studentPortal/labs/multiplication-connect/Index.tsx
Comment (vithushan19) [inline pages/studentPortal/labs/multiplication-connect/Index.tsx:12]: Great job with using descriptive names for your files and components.

It's good to break things up into smaller components. It makes your code easier to manage. Nice work setting things up.

**Review by vithushan19 — COMMENTED** [review]
Nice work so far! Let me know when you need another review.

File: pages/studentPortal/labs/multiplication-connect/Index.tsx
Comment (kavanparam) [reply pages/studentPortal/labs/multiplication-connect/Index.tsx:12]: thanks!

File: pages/studentPortal/labs/multiplication-connect/Index.tsx
Comment (kavanparam) [reply pages/studentPortal/labs/multiplication-connect/Index.tsx:1]: removed

File: components/math/multiplicationConnect/DiceSection.tsx
Comment (kavanparam) [reply components/math/multiplicationConnect/DiceSection.tsx:22]: Yes!

**Comment by vithushan19** [comment]
Since you added more UI changes, could you add more screenshots? It's good practice and helps me to review better

File: components/math/multiplicationConnect/DiceSection.tsx
Comment (vithushan19) [inline components/math/multiplicationConnect/DiceSection.tsx:16]: Nice 👍🏾

File: components/math/multiplicationConnect/GameBoard.tsx
Comment (vithushan19) [inline components/math/multiplicationConnect/GameBoard.tsx:8]: Can we clean this up by creating an array of numbers, then mapping through that array to render these divs?

**Review by vithushan19 — APPROVED** [review]
Nice! Looks good to me, just had one comment around rendering lists better using the map function.

Please fix that before merging this in.

This PR is getting pretty big. It's good practice to keep PRs small. Let's try merging this one in and starting a new branch and PR for your next tasks.

Great work! Feels like you understand how to set state.

**Comment by kavanparam** [comment]
Updated UI

image

File: components/math/multiplicationConnect/DiceSection.tsx
Comment (kavanparam) [reply components/math/multiplicationConnect/DiceSection.tsx:31]: Took a while, but doing my best!

File: components/math/multiplicationConnect/GameBoard.tsx
Comment (kavanparam) [reply components/math/multiplicationConnect/GameBoard.tsx]: Good idea, thank you!

File: components/math/multiplicationConnect/GameBoard.tsx
Comment (vithushan19) [inline components/math/multiplicationConnect/GameBoard.tsx:25]: Instead of calling your helper function you should call `arr.map(...)` here

File: components/math/multiplicationConnect/GameBoard.tsx
Comment (vithushan19) [inline components/math/multiplicationConnect/GameBoard.tsx:5]: This should be a state variable inside your component

File: components/math/multiplicationConnect/GameBoard.tsx
Comment (vithushan19) [inline components/math/multiplicationConnect/GameBoard.tsx:6]: https://stackoverflow.com/questions/3746725/how-to-create-an-array-containing-1-n

File: components/math/multiplicationConnect/GameBoard.tsx
Comment (kavanparam) [reply components/math/multiplicationConnect/GameBoard.tsx:25]: ya this makes more sense

File: components/math/multiplicationConnect/GameBoard.tsx
Comment (kavanparam) [reply components/math/multiplicationConnect/GameBoard.tsx:5]: Good point

File: components/math/multiplicationConnect/GameBoard.tsx
Comment (kavanparam) [reply components/math/multiplicationConnect/GameBoard.tsx:6]: is it necessary to refactor this? I'm getting errors trying to use the following:

```
return Array.from(Array(35), (e,i)=>i+1).map(i => {
            return {
              key: i,
              gridNumber: i,
            }
          });
```

Comment (kavanparam) [reply components/math/multiplicationConnect/GameBoard.tsx:6]: something about React children not being able to have an object and to use an array instead

**Pull request merged**
